string_permutation_problem/permute.py

```diff
@@ -0,0 +1,26 @@
+def permute(s):
+    out = []
+    
+    #Base Case
+    if len(s) == 1:
+        out = [s]
+        
+    else:
+        #For every letter in String
+        for i, let in enumerate(s):
+            
+            #for every permutation resulting from Step 2 and 3 described above
+            
```

Comment:
Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `python spacing`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp3zw2_ifm/string_permutation_problem/permute.py
+++ b/tmp/tmp3zw2_ifm/string_permutation_problem/permute.py
@@ -8,7 +8,7 @@
     else:
         #For every letter in String
         for i, let in enumerate(s):
-            
+
             #for every permutation resulting from Step 2 and 3 described above
             
             for perm in permute(s[:i] + s[i+1:]):
```